ksql-engine/src/test/resources/query-validation-tests/case-expression.json

```diff
@@ -0,0 +1,552 @@
+{
+  "comments": [
+    "You can specify multiple statements per test case, i.e., to set up the various streams needed",
+    "for joins etc, but currently only the final topology will be verified. This should be enough",
+    "for most tests as we can simulate the outputs from previous stages into the final stage. If we",
+    "take a modular approach to testing we can still verify that it all works correctly, i.e, if we",
+    "verify the output of a select or aggregate is correct, we can use simulated output to feed into",
+    "a join or another aggregate."
+  ],
+  "tests": [
+    {
+      "format": ["AVRO", "JSON"],
+      "statements": [
+        "CREATE STREAM orders (ordertime bigint, orderid bigint, itemid STRUCT< ITEMID BIGINT, NAME VARCHAR, CATEGORY STRUCT< ID BIGINT, NAME VARCHAR>>, ORDERUNITS double, ARRAYCOL array<double>, MAPCOL map<varchar, double>, address STRUCT < number bigint, street varchar, city varchar, state varchar, zipcode bigint>) WITH (kafka_topic='test_topic', key='orderid', value_format='{FORMAT}');",
```

Comment:
As requested [before](https://github.com/confluentinc/ksql/pull/2319#discussion_r245961250) can we simplify this schema please?   The only column from the source schema that is used is `orderunits`, so that's the only column we need in there. The others are just noise.   Remove the others to make the test case easier to understand. 

Then you can refactor the inputs into a single line per record.

Reply:
Same goes for the other test cases in this file.